src/script/standard.h

```diff
@@ -90,9 +90,9 @@ struct WitnessV0KeyHash : public uint160
 //! CTxDestination subtype to encode any future Witness version
 struct WitnessUnknown
 {
-    unsigned int version;
-    unsigned int length;
-    unsigned char program[40];
+    unsigned int version = 0;
+    unsigned int length = 0;
+    unsigned char program[40] = {};
```

Comment:
The warning is only reasonable if an uninitialised `WitnessUnknown` is compared with something; that seems a bug in its own right, and not something that should be hidden by auto initialising everything to 0/empty?

How about:

```+    template<typename T>
+    inline WitnessUnknown(unsigned int _version, T _data)
+      : version(_version), length(_data.size())
+    {
+        assert(length <= 40);
+        std::copy(_data.begin(), _data.end(), program);
+    }
+
+    WitnessUnknown() = delete;
```

and changing the various uses from:

```
-            WitnessUnknown unk;
-            unk.version = version;
-            std::copy(data.begin(), data.end(), unk.program);
-            unk.length = data.size();
-            return unk;
```

to just

```
+            return WitnessUnknown(version, data);
```

As far as I can see the warning is a false-positive though?

Reply:
Thinking about it a bit more, that doesn't guarantee the first `length` bytes of `program` are initialised. `std::copy_n(_data, length, program);` might arguably be better.

Reply:
Skipped this change to keep things easy to review. I might post a follow-up PR instead :-)